api/src/test/java/org/openmrs/api/DiagnosisServiceTest.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+
+import org.junit.Test;
+import org.openmrs.Diagnosis;
+import org.openmrs.api.context.Context;
+import org.openmrs.test.BaseContextSensitiveTest;
+
+public class DiagnosisServiceTest extends BaseContextSensitiveTest {
+
+	/**
+	 * Creates a diganosis object.
+	 * @return diagnosis the diagnosis created
+	 */
+	private Diagnosis buildDiagnosis() {
+
+		Diagnosis diagnosis = new Diagnosis();
+		diagnosis.setEncounter(Context.getEncounterService().getEncounter(1));
+		diagnosis.setRank(1);
+		return diagnosis;
+	}
+
+	/**
+	 * @see DiagnosisService#save(Diagnosis)
+	 */
+	@Test
+	public void save_shouldSaveDiagnosisWithBasicDetails() {
+		Diagnosis diagnosis = buildDiagnosis();
+		DiagnosisService ds = Context.getDiagnosisService();
+		ds.save(diagnosis);
+
+		assertNotNull("The saved diagnosis should have an id", diagnosis.getId());
+
+		Diagnosis newDiagnosis = ds.getDiagnosis(diagnosis.getId());
+		assertNotNull("We expect a diagnosis object to be returned", newDiagnosis);
```

Comment:
It's good to check that the uuid matches the expected value since uuids are unique like ids